pkg/arvo/app/settings-store.hoon

```diff
@@ -0,0 +1,167 @@
+/-  *settings
+/+  verb, dbug, default-agent
+|%
++$  card  card:agent:gall
++$  versioned-state
+  $%  state-0
+  ==
++$  state-0
+  $:  %0
+      =settings
+  ==
+--
+=|  state-0
+=*  state  -
+::
+%-  agent:dbug
+%+  verb  |
+^-  agent:gall
+=<
+  |_  bol=bowl:gall
+  +*  this  .
+      do    ~(. +> bol)
+      def   ~(. (default-agent this %|) bol)
+  ::
+  ++  on-init  on-init:def
+  ::
+  ++  on-save  !>(state)
+  ::
+  ++  on-load
+    |=  =old=vase
+    ^-  (quip card _this)
+    =/  old  !<(versioned-state old-vase)
+    ?-  -.old
+      %0  [~ this(state old)]
+    ==
+  ::
+  ++  on-poke
+    |=  [mar=mark vas=vase]
+    ^-  (quip card _this)
+    ?>  (team:title our.bol src.bol)
+    ?.  ?=(%settings-event mar)
+      (on-poke:def mar vas)
+    =/  evt=event  !<(event vas)
+    =^  cards  state
+      ?-  -.evt
+        %put-bucket  (put-bucket:do key.evt bucket.evt)
+        %del-bucket  (del-bucket:do key.evt)
+        %put-entry   (put-entry:do buc.evt key.evt val.evt)
+        %del-entry   (del-entry:do buc.evt key.evt)
+      ==
+    [cards this]
+  ::
+  ++  on-watch
+    |=  pax=path
+    ^-  (quip card _this)
+    ?>  (team:title our.bol src.bol)
+    ?+  pax  (on-watch:def pax)
+        [%all ~]
+      [~ this]
+    ::
+        [%bucket @ ~]
+      =*  bucket-key  i.t.pax
+      ?>  (~(has by settings) bucket-key)
+      [~ this]
+    ::
+        [%entry @ @ ~]
+      =*  bucket-key  i.t.pax
+      =*  entry-key   i.t.t.pax
+      =/  bucket  (~(got by settings) bucket-key)
+      ?>  (~(has by bucket) entry-key)
+      [~ this]
+    ==
+  ::
+  ++  on-peek
+    |=  pax=path
+    ^-  (unit (unit cage))
+    ?+  pax  (on-peek:def pax)
+        [%x %all ~]
+      ``settings-data+!>(all+settings)
+    ::
+        [%x %bucket @ ~]
+      =*  buc  i.t.t.pax
+      =/  bucket=(unit bucket)  (~(get by settings) buc)
+      ?~  bucket  [~ ~]
+      ``settings-data+!>(bucket+u.bucket)
+    ::
+        [%x %entry @ @ ~]
+      =*  buc  i.t.t.pax
+      =*  key  i.t.t.t.pax
+      =/  =bucket  (fall (~(get by settings) buc) ~)
+      =/  entry=(unit val)  (~(get by bucket) key)
+      ?~  entry  [~ ~]
+      ``settings-data+!>(entry+u.entry)
+    ==
+  ::
+  ++  on-agent  on-agent:def
+  ++  on-leave  on-leave:def
+  ++  on-arvo   on-arvo:def
+  ++  on-fail   on-fail:def
+  --
+::
+|_  bol=bowl:gall
```

Comment:
I generally prefer shadowing `=bowl:gall` so as not to be able to accidentally reference the bunted bowl.